base/sparse/sparsematrix.jl

```diff
@@ -942,8 +943,9 @@ function getindex{Tv,Ti<:Integer}(A::SparseMatrixCSC{Tv,Ti}, I::Range, J::Abstra
     # Form the structure of the result and compute space
     for j = 1:nJ
         @inbounds col = J[j]
+        1 <= col <= n || throw(BoundsError())
```

Comment:
The `colptrA` access throws a BoundsError, but since #9534 the error message is more confusing than helpful:

```julia
julia> sparse(zeros(5, 5))[1:2, 5:6]
ERROR: BoundsError: attempt to access 6-element Array{Int64,1}:
 1
 1
 1
 1
 1
 1
  at index [7]
 in getindex at sparse/sparsematrix.jl:49
```

I think the extra `@inbounds` should be fine.